src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp

```diff
@@ -1958,8 +1955,214 @@ void jit_soft_sign_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, con
+                                      const emitter_context *emitContext) const {
+    if (host_isa_ == x64::avx512_core) {
+        emit_isa<x64::avx512_core>(inVecIdxs, outVecIdxs);
+    } else if (host_isa_ == x64::avx2) {
+        emit_isa<x64::avx2>(inVecIdxs, outVecIdxs);
+    } else if (host_isa_ == x64::sse41) {
+        emit_isa<x64::sse41>(inVecIdxs, outVecIdxs);
+    } else {
+        assert(!"Unsupported isa");
+    }
+}
+
+/// IS_INF ///
+template <>
+void jit_is_inf_emitter::emit_isa<x64::avx512_core>(const std::vector<size_t> &inVecIdxs, const std::vector<size_t> &outVecIdxs) const {
+    Zmm vSrc = Zmm(inVecIdxs[0]);
+    Zmm vDst = Zmm(outVecIdxs[0]);
+
+    if (detectNegative || detectPositive) {
+        auto &onesMask = h->k1;
+        auto rOne = Reg32(aux_gpr_idxs[0]);
+        uint8_t imm = detectNegative ? 0B00010000 : 0B00000000;
+        if (detectPositive) {
+            imm |= 0B00001000;
+        }
+
+        h->mov(rOne, const_1F);
+        h->vfpclassps(onesMask, vSrc, imm);
+        h->vpbroadcastd(vDst | onesMask | h->T_z, rOne);
+    } else {
+        h->uni_vxorps(vDst, vDst, vDst);
+    }
+}
+
+template <x64::cpu_isa_t isa>
+void jit_is_inf_emitter::emit_isa(const std::vector<size_t> &inVecIdxs, const std::vector<size_t> &outVecIdxs) const {
+    using Vmm = typename conditional<isa == x64::sse41, Xmm, Ymm>::type;
+    Vmm vSrc = Vmm(inVecIdxs[0]);
+    Vmm vDst = Vmm(outVecIdxs[0]);
+
+    if (detectNegative || detectPositive) {
+        auto rAux = Reg32(aux_gpr_idxs[0]);
+
+        if (detectPositive) {
+            auto vAux = Vmm(aux_vec_idxs[0]);
+            auto xmmAux = Xmm(vAux.getIdx());
+
+            h->mov(rAux, 0x7F800000); // +Inf mask
+            h->uni_vpinsrd(xmmAux, xmmAux, rAux, 0x0);
+            h->uni_vpbroadcastd(vAux, xmmAux);
+            if (isa == x64::avx2) {
+                h->uni_vandps(vDst, vSrc, vAux);
+            } else {
+                h->uni_vmovups(vDst, vSrc);
+                h->uni_vandps(vDst, vDst, vAux);
+            }
+            h->uni_vpcmpeqd(vDst, vDst, vAux);
+
+            if (!detectNegative) {
+                h->mov(rAux, 0xFF800000); // -Inf mask
+                h->uni_vpinsrd(xmmAux, xmmAux, rAux, 0x0);
+                h->uni_vpbroadcastd(vAux, xmmAux);
+                h->uni_vandps(vSrc, vSrc, vAux);
+                h->uni_vpcmpeqd(vSrc, vSrc, vAux);
+                if (isa == x64::avx2) {
+                    h->vandnps(vDst, vSrc, vDst);
+                } else {
+                    h->andnps(vSrc, vDst);
+                    h->uni_vmovups(vDst, vSrc);
+                }
+            }
+
+            h->mov(rAux, const_1F);
+            h->uni_vpinsrd(xmmAux, xmmAux, rAux, 0x0);
+            h->uni_vpbroadcastd(vAux, xmmAux);
+            h->uni_vandps(vDst, vDst, vAux);
+        } else if (detectNegative) {
+            auto xmmDst = Xmm(vDst.getIdx());
+
+            h->mov(rAux, 0xFF800000); // NaN mask
```

Comment:
Could you verify please that it's really Nan mask? Seems like it's `-inf mask` so maybe we should update comment?

Reply:
Done